torch_xla/csrc/runtime/runtime.cc

```diff
@@ -8,35 +8,29 @@
 
 namespace torch_xla {
 namespace runtime {
-namespace {
 
 std::atomic<bool> g_computation_client_initialized(false);
 
-ComputationClient* CreateClient() {
-  bool was_initialized = g_computation_client_initialized.exchange(true);
-  XLA_CHECK(!was_initialized) << "ComputationClient already initialized";
-  if (sys_util::GetEnvBool("XLA_DUMP_FATAL_STACK", false)) {
-    tsl::testing::InstallStacktraceHandler();
-  }
-
-  ComputationClient* client;
+ComputationClient* GetComputationClient() {
+  static std::unique_ptr<ComputationClient> client = []() {
```

Comment:
What's the difference between this and the original approach, since both rely on the static initializer? It seems like the logic has just been moved out of CreateClient and into the lambda.

Reply:
Yeah, that's right. This lets me skip checking the case where this function gets called twice, the handling of which causes the bad error message. C++11 statics will ensure that it only completes once, and the function is anonymous now, which will prevent future code from erroneously calling `CreateClient` somewhere else in the future.

Reply:
Ah I see, the lambda is guaranteed to only ever be called once. Do we know how `CreateClient` was called twice in the first place?

Reply:
I added that check to defend against future errors (likely by future me). There was never a case in the original code where `CreateClient` actually _completed_ twice.

The bad error messaging occurred when the constructor of `PjRtComputationClient` threw an exception. `g_computation_client_initialized` was never reset to `false`, `GetComputationClientIfInitialized` calls `GetComputationClient` during teardown, which re-runs `CreateClient`, which complains because `g_computation_client_initialized` never got reset.

In this PR, I only set `g_computation_client_initialized` _after_ the actual runtime init completed without breaking, since there are guaranteed to be no other concurrent callers of `CreateClient`.